tests/suites/test_suite_x509write.function

```diff
@@ -549,14 +558,42 @@ exit:
     mbedtls_pk_free(&issuer_key_alt);
     mbedtls_pk_free(&subject_key);
     mbedtls_pk_free(&issuer_key);
-    mbedtls_mpi_free(&serial);
+#if defined(MBEDTLS_TEST_DEPRECATED) && defined(MBEDTLS_BIGNUM_C)
+    mbedtls_mpi_free(&serial_mpi);
+#endif
 #if defined(MBEDTLS_USE_PSA_CRYPTO)
     psa_destroy_key(key_id);
 #endif
     USE_PSA_DONE();
 }
 /* END_CASE */
 
+/* BEGIN_CASE depends_on:MBEDTLS_X509_CRT_WRITE_C */
+void x509_set_serial_check()
```

Comment:
I would suggest extending this test to be able to: 
- pass serial length as an argument;
- pass expected return code of `mbedtls_x509write_crt_set_serial`.
This way we can test

Reply:
But aren't valid lengths already tested in `x509_crt_check()` above?

Reply:
But not using `mbedtls_x509write_crt_set_serial_raw`? I can see that the new code added in `mbedtls_x509write_crt_set_serial_raw` (but with a valid serial_len) is not tested anywhere.

Reply:
Sorry, but I still don't see what's missing... :(
To summarize:

- "old" function `mbedtls_x509write_crt_set_serial()` is tested for
    - valid serials [here](https://github.com/valeriosetti/mbedtls/blob/18b9b035ada7a1fd63835bc7d7f1adfb37de44b8/tests/suites/test_suite_x509write.function#L398)
    - invalid serials [here](https://github.com/valeriosetti/mbedtls/blob/18b9b035ada7a1fd63835bc7d7f1adfb37de44b8/tests/suites/test_suite_x509write.function#L585)

- "new" function `mbedtls_x509write_crt_set_serial_raw()` is tested for
    - valid serials [here](https://github.com/valeriosetti/mbedtls/blob/18b9b035ada7a1fd63835bc7d7f1adfb37de44b8/tests/suites/test_suite_x509write.function#L400)
    - invalid serials [here](https://github.com/valeriosetti/mbedtls/blob/18b9b035ada7a1fd63835bc7d7f1adfb37de44b8/tests/suites/test_suite_x509write.function#L589)

It seems to me that also the new function is tested. Actually `mbedtls_x509write_crt_set_serial_raw` only performs 1 check internally, so once valid and invalid length serials are tested this should be enough. Isn't it?

Reply:
Ah, you're right, I ran the coverage test without `MBEDTLS_DEPRECATED_REMOVED`.  My bad!